src/Pos/Generator/Block/Mode.hs

```diff
@@ -191,20 +195,25 @@ instance HasSlottingVar InitBlockGenContext where
     slottingVar = ibgcSlottingVar_L
 
 instance MonadBlockGenBase m => MonadDBRead (InitBlockGenMode m) where
-    dbGet = DB.dbGetPureDefault
-    dbIterSource = DB.dbIterSourcePureDefault
+    dbGet tag key = eitherDB (DB.dbGetDefault tag key) (DB.dbGetPureDefault tag key)
+    dbIterSource tag proxy = view (lensOf @GS.DBSum) >>= \case
+        GS.RealDB dbs -> hoist (flip runReaderT dbs) (DB.dbIterSourceDefault tag proxy)
+        GS.PureDB pdb -> hoist (hoist $ flip runReaderT pdb) (DB.dbIterSourcePureDefault tag proxy)
```

Comment:
No, I don't like that this logic is here. It should be moved to `db`. There should be `DBSum` type and `dbGetSum` and `dbIterSourceSum` methods. Same for other DB classes.